test.html

```diff
@@ -56,45 +56,45 @@
     <!--color-->
     <fieldset class="pure-group">
       <label for="color">Favourite Color: </label>
-      <input type="color" name="color" />
+      <input id="color" type="color" name="color" />
     </fieldset>
 
     <!--radio button-->
     <fieldset class="pure-group">
       <label for="radio">Choose a Programming Language: </label>
-      <input type="radio" name="radio" value="js" checked />Javascript
+      <input id="radio" type="radio" name="radio" value="js" checked />Javascript
     </fieldset>
 
     <!--radio group-->
     <fieldset class="pure-group">
-      <label for="radio-group">OK, Now Really Choose:</label>
-      <input type="radio" name="radio-group" value="js" />Javascript
-      <input type="radio" name="radio-group" value="java" />Java
-      <input type="radio" name="radio-group" value="c++" />C++
-      <input type="radio" name="radio-group" value="python" checked />Python
-      <input type="radio" name="radio-group" value="other" />Other
+      <legend>OK, Now Really Choose:</legend>
+      <input id="radio-group--js" type="radio" name="radio-group" value="js" /> <label for="radio-group--javascript">Javascript</label>
```

Comment:
Just a minor bug in the `id` vs what's in the `for` here, everything else seems to work fine with the labels!